builtin/providers/openstack/resource_openstack_image_v2.go

```diff
@@ -0,0 +1,494 @@
+				ForceNew: true,
+			},
+
+			"owner": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"checksum": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"size_bytes": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"metadata": &schema.Schema{
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+
+			"created_date": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"last_update": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"file": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"schema": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func validateVisibility(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(images.ImageVisibility)
+	if value == images.ImageVisibilityPublic || value == images.ImageVisibilityPrivate || value == images.ImageVisibilityShared || value == images.ImageVisibilityCommunity {
+		return
+	}
+	errors = append(errors, fmt.Errorf("%q must be one of %q, %q, %q, %q", k, images.ImageVisibilityPublic, images.ImageVisibilityPrivate, images.ImageVisibilityCommunity, images.ImageVisibilityShared))
+	return
+}
+
+func validatePositiveInt(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(int)
+	if value > 0 {
+		return
+	}
+	errors = append(errors, fmt.Errorf("%q must be a positive integer", k))
+	return
+}
+
+var DiskFormats = [9]string{"ami", "ari", "aki", "vhd", "vmdk", "raw", "qcow2", "vdi", "iso"}
+
+func validateDiskFormat(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	for i := range DiskFormats {
+		if value == DiskFormats[i] {
+			return
+		}
+	}
+	errors = append(errors, fmt.Errorf("%q must be one of %v", k, DiskFormats))
+	return
+}
+
+var ContainerFormats = [9]string{"ami", "ari", "aki", "bare", "ovf"}
+
+func validateContainerFormat(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	for i := range ContainerFormats {
+		if value == ContainerFormats[i] {
+			return
+		}
+	}
+	errors = append(errors, fmt.Errorf("%q must be one of %v", k, ContainerFormats))
+	return
+}
+
+func imageV2VisibitlityFromString(v string) (images.ImageVisibility, error) {
+	if images.ImageVisibility(v) == images.ImageVisibilityCommunity {
+		return images.ImageVisibilityCommunity, nil
+	}
+	if images.ImageVisibility(v) == images.ImageVisibilityPrivate {
+		return images.ImageVisibilityPrivate, nil
+	}
+	if images.ImageVisibility(v) == images.ImageVisibilityPublic {
+		return images.ImageVisibilityPublic, nil
+	}
+	if images.ImageVisibility(v) == images.ImageVisibilityShared {
+		return images.ImageVisibilityShared, nil
+	}
+
+	return "", fmt.Errorf("Error unkown image visibility %q", v)
+}
```

Comment:
nit: I think a `switch` statement might be a better approach. See [here](https://github.com/jtopjian/terraform/blob/openstack-data-source-image/builtin/providers/openstack/data_source_openstack_images_image.go#L239) for an example.